src/box/box.cc

```diff
@@ -174,6 +174,21 @@ box_update_ro_summary(void)
 	fiber_cond_broadcast(&ro_cond);
 }
 
+const char *
+box_ro_reason(void)
+{
+	if (raft_is_ro(box_raft()))
+		return "election";
+	if (txn_limbo_is_ro(&txn_limbo))
+		return "synchro";
+	if (is_ro)
+		return "config";
+	if (is_orphan)
+		return "orphan";
+	assert(false);
```

Comment:
I'd remove this assertion and call `box_ro_reason` from `lbox_info_ro_reason` unconditionally (without checking `box_is_ro`) so that `box_ro_reason` is self-sufficient, but this is nit-picking. Up to you. The patch LGTM.